bentoml/_internal/server/runner_app.py

```diff
@@ -181,23 +180,26 @@ async def _run(requests: t.Iterable[Request]) -> list[Response]:
                 *tuple(multipart_to_payload_params(r) for r in requests)
             )
 
-            batch_dim = runner_method.runnable_method_config.batch_dim
-            mix_params = Params.agg(
-                params_list=params_list,
-                agg_func=lambda i: AutoContainer.from_batch_payloads(
-                    i, batch_dim=batch_dim[0]
-                ),
-            )
-            batched_params, indices_params = mix_params.vsplit(lambda i: (i[0], i[1]))
+            batch_dim = runner_method.config.batch_dim
+
+            try:
+                batched_params, indices = batch_params(params_list, batch_dim[0])
```

Comment:
we actually do need that batch_params, I think. How about just checking by iterating on the indices_params?

Reply:
This refactoring makes general container utils on Pramas circumstance specified, with more cupling.